driver/layernorm_driver.hpp

```diff
@@ -0,0 +1,429 @@
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    std::vector<int> GetInputTensorLengthsFromCmdLine();
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~LayerNormDriver() override
+    {
+
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(weightDesc);
+        miopenDestroyTensorDescriptor(biasDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+        miopenDestroyTensorDescriptor(meanDesc);
+        miopenDestroyTensorDescriptor(rstdDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+    int dim_size;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t weightDesc;
+    miopenTensorDescriptor_t biasDesc;
+    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t meanDesc;
+    miopenTensorDescriptor_t rstdDesc;
+
+    std::unique_ptr<GPUMem> in_dev;
+    std::unique_ptr<GPUMem> weight_dev;
+    std::unique_ptr<GPUMem> bias_dev;
+    std::unique_ptr<GPUMem> out_dev;
+    std::unique_ptr<GPUMem> mean_dev;
+    std::unique_ptr<GPUMem> rstd_dev;
+
+    std::vector<Tgpu> in;
+    std::vector<Tgpu> weight;
+    std::vector<Tgpu> bias;
+    std::vector<Tgpu> out;
+    std::vector<Tgpu> mean;
+    std::vector<Tgpu> rstd;
+    std::vector<Tref> outhost;
+    std::vector<Tref> meanhost;
+    std::vector<Tref> rstdhost;
+
+    float eps;
+    int dim;
+    miopenLayerNormMode_t mode;
+};
+
+template <typename Tgpu, typename Tref>
+int LayerNormDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int LayerNormDriver<Tgpu, Tref>::GetandSetData()
+{
+    std::vector<int> in_len = GetInputTensorLengthsFromCmdLine();
+
+    dim = static_cast<int>(inflags.GetValueDouble("nomalized_dim"));
+
+    std::vector<int> inner_len;
+    if(dim == in_len.size())
+        inner_len = {1};
+    else
+        inner_len = {in_len.begin() + dim, in_len.end()};
+
+    std::vector<int> outer_len;
+    if(dim == 0)
+        outer_len = {1};
+    else
+        outer_len = {in_len.begin(), in_len.end() - (in_len.size() - dim)};
+
+    SetTensorNd(inputDesc, in_len, data_type);
+    SetTensorNd(weightDesc, inner_len, data_type);
+    SetTensorNd(biasDesc, inner_len, data_type);
+    SetTensorNd(outputDesc, in_len, data_type);
+    SetTensorNd(meanDesc, outer_len, data_type);
+    SetTensorNd(rstdDesc, outer_len, data_type);
+
+    eps  = static_cast<double>(inflags.GetValueDouble("eps"));
+    mode = miopenLayerNormMode_t(inflags.GetValueInt("mode"));
+
+    return (0);
```

Comment:
[future][R] this is legacy junk what I would like to avoid in the new designs
```suggestion
    return 0;
```